src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py

```diff
@@ -1492,7 +1498,7 @@ def load_annulus_description(
         "pvt_model": load_value,
         "top_node": load_value,
         "initial_conditions": load_initial_conditions_description,
-        "gas_lift_valve_equipment": load_gas_lift_valve_equipment_description,
+        "equipment": partial(load_instance, class_=case_description.AnnulusEquipmentDescription),
```

Comment:
```suggestion
        "equipment": get_instance_loader(class_=case_description.AnnulusEquipmentDescription),
```
If not using auto loaders at least keep it simple as possible.

Reply:
isn't that "using auto loaders" thou?

Reply:
it uses part of the autoloaders infra.
Those dicts are explicit loaders:
```
explicit_loaders_for_DecriptionType = {
    "attr_name_FOO": load_vale,
    "attr_name_BAR": load_BAR_description,
    "attr_name_XXX": get_instance_loader(class_=case_description.XXXDescription),
}
```

could be written as:
```
explicit_loaders = {
    "attr_name_BAR": load_BAR_description,
}
loaders = get_case_description_attribute_loader_dict(DecriptionType, explicit_loaders)
```

`load_value` it the default loader when no explicit loader is supplied and no metadata is present.
If "attr_name_BAR" is also declared with have metadata (and the autloader is exits) no explicit loader is required and  possibly we can remove `load_BAR_description` altogether.

Reply:
I would suguest, first just try

```
def load_annulus_description(
    document: DescriptionDocument,
) -> case_description.AnnulusDescription:
    return load_instance(document, case_description.AnnulusDescription)
```
If this works you probably can remove `load_initial_conditions_description` if its no longer use (probably is be the main pipe loader).

If the full autloader do not work try:

```
# Change `load_instance` to accept optional explict loaders.
def load_instance(alfacase_content: DescriptionDocument, class_: Type[T], explicit_loaders: Optional[Dict[str, Callable]] = None) -> T:
    """
    Create an instance of class_ with the attributes found in alfacase_content.
    """
    alfacase_to_case_description = get_case_description_attribute_loader_dict(class_, explicit_loaders)
    case_values = to_case_values(alfacase_content, alfacase_to_case_description)
    item_description = class_(**case_values)
    return update_multi_input_flags(alfacase_content, item_description)


def load_annulus_description(
    document: DescriptionDocument,
) -> case_description.AnnulusDescription:
    explicit_loaders= {
        "initial_conditions": load_initial_conditions_description,
    }
    return load_instance(document, case_description.AnnulusDescription, explicit_loaders)
```

Reply:
BTW, `explicit_loaders` are not actually used today. I did add it to help with progressive adoption of the autoloaders infrastructure.
I did use if in the initial port of the loaders already ported.

1. Update the loader to use the autoloaders infrastructure with explict loaders;
2. Add necessary metadata and autoloaders support;
3. When the explict loader dict is empty just delete the function =);

This allow to keep the tests passing while progressively implementing the loaders. Implement 1, run the tests, another, run the test. This makes if easier to track a problem when implementing.